src/test/FuzzerImpl.cpp

```diff
@@ -503,17 +517,82 @@ isBadOverlayFuzzerInput(StellarMessage const& m)
 
 static void
 attemptToApplyOps(LedgerTxn& ltx, PublicKey const& sourceAccount,
-                  xdr::xvector<Operation> const& ops, Application& app)
+                  xdr::xvector<Operation> const& ops, Application& app,
+                  bool const throwIfTxFails = true)
 {
     auto txFramePtr =
         createFuzzTransactionFrame(sourceAccount, ops, app.getNetworkID());
     txFramePtr->attemptApplication(app, ltx);
 
-    if (txFramePtr->getResultCode() != txSUCCESS)
+    if (throwIfTxFails)
     {
-        throw std::runtime_error(fmt::format(
-            FMT_STRING("Error {} while applying operations while fuzzing"),
-            txFramePtr->getResultCode()));
+        auto const opFrames = txFramePtr->getOperations();
+        if (txFramePtr->getResultCode() != txSUCCESS)
+        {
+            auto const msg = fmt::format(
+                FMT_STRING("Error {} while applying operations while fuzzing"),
+                txFramePtr->getResultCode());
+            LOG_FATAL(DEFAULT_LOG, "{}", msg);
+            for (auto const& op : opFrames)
+            {
+                LOG_ERROR(DEFAULT_LOG, "op {}: result code {}",
+                          op->getOperation().body._xdr_discriminant(),
+                          op->getResultCode());
+            }
+            throw std::runtime_error(msg);
+        }
+        for (auto const& op : opFrames)
+        {
+            if ((op->getOperation().body.type() == MANAGE_BUY_OFFER &&
```

Comment:
This is pretty unpleasant to look at. Use references (similar to what I was saying in the comment on 566) to `op->getOperation().body` and `op->getResult().tr()` to trim this down.

Reply:
Fixed as in previous comment.